framework/src/modules/interoperability/mainchain/commands/register_sidechain.ts

```diff
@@ -116,14 +113,22 @@ export class RegisterSidechainCommand extends BaseInteroperabilityCommand<Mainch
 			};
 		}
 
+		// Chain ID cannot be the mainchain chain ID.
+		if (chainID.equals(context.chainID)) {
```

Comment:
context.chainID is the chainID of itself, it should use the util function to compute the mainchain ID

Reply:
`context.chainID` is already the mainchainID, so there is no need to compute the mainchain ID